lib/mapObjects/MiscObjects.cpp

```diff
@@ -1394,11 +1394,16 @@ void CGArtifact::serializeJsonOptions(JsonSerializeFormat& handler)
 
 void CGWitchHut::initObj(CRandomGenerator & rand)
 {
-	if (allowedAbilities.empty()) //this can happen for RMG. regular maps load abilities from map file
+	if (allowedAbilities.empty()) //this can happen for RMG and RoE maps.
 	{
-		// Necromancy can't be learned on random maps
-		for(int i = 0; i < VLC->skillh->size(); i++)
-			if(VLC->skillh->getByIndex(i)->getId() != SecondarySkill::NECROMANCY)
+		auto defaultAllowed = VLC->skillh->getDefaultAllowed();
+
+		// Necromancy and Leadership can't be learned by default
+		defaultAllowed[SecondarySkill::NECROMANCY] = false;
+		defaultAllowed[SecondarySkill::LEADERSHIP] = false;
```

Comment:
Are Resistance and Interference secondary skill taken into account?

Reply:
@krs0 but our engine is not ever know about interference (it is a HoTA mod thing).

Reply:
Have yet to investigate how Resistance/Interference switch works.

Need to recheck but I think both skills are listed as "allowed" in h3m. In that case it means that this switch is hardcoded in hota exe (ugh...) and will have to handle it in some way. Probably in hota interference submod (will have to extract it from cove submod)

That being said, it looks like witch hut does not checks for skills disabled on map (both old and new code). Will fix.